awx_collection/plugins/modules/tower_job_template.py

```diff
@@ -12,409 +13,532 @@
+        new_fields['skip_tags'] = skip_tags
+    if start_at_task is not None:
+        new_fields['start_at_task'] = start_at_task
+    if timeout is not None:
+        new_fields['timeout'] = timeout
+    if use_fact_cache is not None:
+        new_fields['use_fact_cache'] = use_fact_cache
+    if host_config_key is not None:
+        new_fields['host_config_key'] = host_config_key
+    if ask_scm_branch_on_launch is not None:
+        new_fields['ask_scm_branch_on_launch'] = ask_scm_branch_on_launch
+    if ask_diff_mode_on_launch is not None:
+        new_fields['ask_diff_mode_on_launch'] = ask_diff_mode_on_launch
+    if ask_variables_on_launch is not None:
+        new_fields['ask_variables_on_launch'] = ask_variables_on_launch
+    if ask_limit_on_launch is not None:
+        new_fields['ask_limit_on_launch'] = ask_limit_on_launch
+    if ask_tags_on_launch is not None:
+        new_fields['ask_tags_on_launch'] = ask_tags_on_launch
+    if ask_skip_tags_on_launch is not None:
+        new_fields['ask_skip_tags_on_launch'] = ask_skip_tags_on_launch
+    if ask_job_type_on_launch is not None:
+        new_fields['ask_job_type_on_launch'] = ask_job_type_on_launch
+    if ask_verbosity_on_launch is not None:
+        new_fields['ask_verbosity_on_launch'] = ask_verbosity_on_launch
+    if ask_inventory_on_launch is not None:
+        new_fields['ask_inventory_on_launch'] = ask_inventory_on_launch
+    if ask_credential_on_launch is not None:
+        new_fields['ask_credential_on_launch'] = ask_credential_on_launch
+    if survey_enabled is not None:
+        new_fields['survey_enabled'] = survey_enabled
+    if become_enabled is not None:
+        new_fields['become_enabled'] = become_enabled
+    if diff_mode is not None:
+        new_fields['diff_mode'] = diff_mode
+    if allow_simultaneous is not None:
+        new_fields['allow_simultaneous'] = allow_simultaneous
+    if custom_virtualenv is not None:
+        new_fields['custom_virtualenv'] = custom_virtualenv
+    if job_slice_count is not None:
+        new_fields['job_slice_count'] = job_slice_count
+    if webhook_service is not None:
+        new_fields['webhook_service'] = webhook_service
+    if webhook_credential is not None:
+        new_fields['webhook_credential'] = webhook_credential_id
+
+    if state == 'absent':
+        # If the state was absent we can let the module delete it if needed, the module will handle exiting from this
+        module.delete_if_needed(existing_item)
+    elif state == 'present':
+        # If the state was present and we can let the module build or update the existing item, this will return on its own
+        module.create_or_update_if_needed(
+            existing_item, new_fields,
+            endpoint='job_templates', item_type='job_template',
+            associations={
+                'credentials': credentials_ids,
+                'labels': labels_ids,
+            },
+            additional_posts=additional_posts,
```

Comment:
To get a sense of how this compares to the current state, I wrote this, partially influenced by your code:

https://github.com/ansible/awx/pull/6291/commits/581d495f66ee831ffbf4f61422664a5e139c8886

I'm trying to grasp how your PR winds up with much more code, and it may be largely in the error handling.